In [1]:
from movielens import MovieLens
import pandas as pd

In [2]:
lens = MovieLens("neo4j")

### Find out movies' name which contains "pacific" (Case in-sensitive) 

In [14]:
pd.DataFrame(lens.find_movies_by_name("pacific"))

movieId                                              title
0   25878                          Across the Pacific (1942)
1    3143                         Hell in the Pacific (1968)
2   90260              Hotel Pacific (Zaklete rewiry) (1975)
3    6435                           Operation Pacific (1951)
4    3219                             Pacific Heights (1990)
5  103228                                 Pacific Rim (2013)
6  129843  Plastic Paradise: The Great Pacific Garbage Pa...
7   87435  Sergeant Körmy and the South Pacific (Vääpeli ...
8    2941                               South Pacific (1958)
9  126430                                 The Pacific (2010)

In [ ]:
### Find out movies' name which contains "Phenomenon" (Case sensitive) 

In [15]:
pd.DataFrame(lens.find_movies_by_name("Phenomenon", case_insensitive=True))

movieId              title
0     802  Phenomenon (1996)

# Find a movie with exact ID 
###  What is the movie name of which has movieId as 6000?

In [13]:
pd.DataFrame(lens.find_movie_by_id("296"))

average_rate  count movieId                title
0      4.174231  67310     296  Pulp Fiction (1994)

So average rate is 4.17, and total number of rating is 67,310. 

# Recommendation

### I like Pulp fiction (296), Phenomenon (802), What movie you can recommend it to me?

In [12]:
pd.DataFrame(lens.recommendation(['296', '802'], rating=4.5, limit=5))

average_rate  count movieId  \
0      4.576798  19727     318   
1      4.519893  17343      50   
2      4.599935  12253     858   
3      4.562500      8  106503   
4      4.500000      7  101850   
5      4.583333      6   26416   
6      4.800000      5   78064   
7      4.800000      5   53774   
8      4.700000      5   65357   
9      4.700000      5   33709   

                                               title  
0                   Shawshank Redemption, The (1994)  
1                         Usual Suspects, The (1995)  
2                              Godfather, The (1972)  
3                      Enemies of Reason, The (2007)  
4           Death on the Staircase (Soupçons) (2004)  
5                                    Best Boy (1979)  
6                        Ween Live in Chicago (2004)  
7  Mystery of Picasso, The (Le mystère Picasso) (...  
8               House of Sand (Casa de Areia) (2005)  
9                               Toto le héros (1991)

Based on users who give good rates to "Pulp finction" & "Phenomenon" (both starred by John Travolta), 
we can search other movies which they give generous score already. 

So *Shawshank Redemption*, *Usual Suspects*, *Godfather* is top 3 recommendation for those people, like me. 